kad/kad.go

```diff
@@ -28,3 +28,40 @@ type Key[K any] interface {
 	// and 0 if both keys are equal.
 	Compare(other K) int
 }
+
+// RoutingTable is the interface all Kademlia Routing Tables types support.
+type RoutingTable[K Key[K]] interface {
+	// Self returns the local node's Kademlia key
+	Self() K
+
+	// AddPeer tries to add a peer to the routing table
+	AddPeer(NodeID[K]) (bool, error)
+
+	// RemoveKey tries to remove a peer identified by its Kademlia key from the
+	// routing table
+	RemoveKey(K) (NodeID[K], error)
```

Comment:
I changed the return signature from `bool` to `NodeID[K]`.

Reply:
Why the change? How do we indicate that the key was not removed because it was not in the routing table.

Reply:
yeah, stupid, let's keep the bool 👍